lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -714,6 +724,20 @@ def get_launch_object(connection, ec2_connection):
             launch_object = {"LaunchTemplate": {"LaunchTemplateId": lt['LaunchTemplateId'], "Version": launch_template['version']}}
         else:
             launch_object = {"LaunchTemplate": {"LaunchTemplateId": lt['LaunchTemplateId'], "Version": str(lt['LatestVersionNumber'])}}
+        launch_object = {'LaunchTemplate': launch_object}
```

Comment:
This is unnecessary, it's creating an object that looks like: launch_object={'LaunchTemplate': {'LaunchTemplate': {'LaunchTemplateId': 'lt-02f29cf111d929cbb', 'Version': '1'}}}